### ISSUE TITLE
ENH: Don't run masking on setup scans

### ISSUE THREAD

**Issue by CJ-Wright** [xpdAcq/xpdAn#152]

**Comment by CJ-Wright** [comment]
Basic workflow write a `is_setup` predicate, if not setup route through the normal masking setup, otherwise run it through a `mask=np.ones(img.shape)` node.

**Comment by sbillinge** [comment]
I am still torn on this masking issue.  I would actually like dynamic vs. static masking to be a user-selectable preference even for non-setup runs.  Speed is one issue, but also users getting consistent results and there not being too much "magic" in the data analysis pipeline that they can't explain in their papers.  From this perspective, I am much more comfortable with a workflow where a mask is created and then re-used for a "set" of measurements and somehow stored.  I can then reproduce results exactly by using the same mask, and then deliberately change the mask if I am not happy with the results.

I am not against offering dynamic masking to users as an option, but I am still uncomfortable with having it as the default.

We could give it similar behavior to dark correction, where if it can't find a "fresh" mask it will generate a dynamic one.

**Comment by CJ-Wright** [comment]
I don't think that this is too magical:
1. We are consistent when we produce a new mask (every run of `xrun`, or potentially every non-setup run of `xrun` if this issue is accepted).
1. The mask will consistently return the same statistical results (pixels who's value is greater than 3 standard deviations away from the mean will be masked out). This gives us some provenance, since the users can ask why a given pixel was removed, and we can return "because it was x standard deviations away from the mean", rather than just "because we masked it on a previous image". 
1. We always write the mask down to disk with the corresponding file name to the data, allowing users to reuse the mask as needed and reproduce their exact results. The users can then take their data home to either use that mask, use a different auto mask, make a new one from scratch, or edit the existing mask.
1. Other software platforms have started to do this as well (although they don't write masks they just remove the pixels before integration, see pyFAI sigma clipping).
1. There is a paper describing how all this works, while I don't think that the ideal mask generating sample has been explored or published. I imagine that the users papers (if they chose to use the auto mask) would have a line saying: "A mask was produced using the automask procedure [citation] with a standard deviation threshold of 3". If the users then provided the data then others could set their masking to the exact same parameters to fully reproduce the mask, which is simpler than trying to recreate a user defined mask.

To some extent I would say that not using automated masking, or running with a mask which was previously produced is _more_ magical (sometimes things will work, sometimes things won't and it won't be obvious why that happened). For some images the previous mask will produce data which has all of the "3 standard deviation or greater" pixels removed, but for others it will not since the scattering is different.

I understand the time based concerns with setup scans, they exist to check if the data is reasonable before taking more "production" quality shots.

Making the mask generation dynamic could be rather difficult to implement, since once the data processing pipeline is running there is no current way to change it's parameters (although we may be able to put something into the experimental metadata?)
